runtime/parachains/src/paras.rs

```diff
@@ -300,21 +366,68 @@ impl<T: Config> Module<T> {
 	fn apply_incoming(parachains: &mut Vec<ParaId>) {
 		let upcoming = <Self as Store>::UpcomingParas::take();
 		for upcoming_para in upcoming {
+			let state = match ParaLifecycles::get(&upcoming_para) {
+				Some(ParaLifecycle::OnboardingAsParachain) => ParaLifecycle::OnboardingAsParachain,
+				Some(ParaLifecycle::OnboardingAsParathread) => ParaLifecycle::OnboardingAsParathread,
+				_ => continue,
+			};
+
 			let genesis_data = match <Self as Store>::UpcomingParasGenesis::take(&upcoming_para) {
```

Comment:
why is the information contained in here being duplicated in `ParaLifecycle`? Why not collapse both onboarding enum items into just `ParaLifecycle::Onboarding` and let `UpcomingParasGenesis` determine whether it is a parachain/parathread?

Reply:
I have collapsed the onboarding state 👍